### PR TITLE
Extract repositories metrics into its own class

### PR DESCRIPTION
**Pull request by ywangd**
This PR is a follow up of
https://github.com/elastic/elasticsearch/pull/102505#discussion_r1402957598 that move the repositories metrics management into its own class which is then passed around instead of relying on the raw meterRegistry and string metric names.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/repositories/RepositoriesMetrics.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    private final LongCounter requestCounter;
+
+    private final LongCounter exceptionCounter;
+    private final LongCounter throttleCounter;
+    private final LongCounter operationCounter;
+    private final LongCounter unsuccessfulOperationCounter;
+    private final LongHistogram exceptionHistogram;
+    private final LongHistogram throttleHistogram;
+
+    public RepositoriesMetrics(MeterRegistry meterRegistry) {
```
Comment (ywangd) [inline server/src/main/java/org/elasticsearch/repositories/RepositoriesMetrics.java:36]: This is the new class. Everything else is cascading changes

**Comment by ywangd** [comment]
@JVerwolf What is the status of https://github.com/elastic/elasticsearch/pull/102435? Do you want me to hold this PR till yours is over? Thanks!

File: server/src/main/java/org/elasticsearch/repositories/RepositoriesMetrics.java (hunk 1 above)
Comment (idegtiarenko) [reply server/src/main/java/org/elasticsearch/repositories/RepositoriesMetrics.java:36]: Considering its structure is there a value in converting this class to a record and replace constructor with a static factory?
This way we will not need to declare getters

**Review by idegtiarenko — APPROVED** [review]
Left optional suggestion, otherwise :+1:

File: server/src/main/java/org/elasticsearch/repositories/RepositoriesMetrics.java (hunk 1 above)
Comment (ywangd) [reply server/src/main/java/org/elasticsearch/repositories/RepositoriesMetrics.java:35]: Sure I changed it to be a record.

**Comment by ywangd** [comment]
@elasticmachine run elasticsearch-ci/part-3

**Pull request merged**